presto-main/src/main/java/com/facebook/presto/operator/scalar/MathFunctions.java

```diff
@@ -1101,11 +1103,17 @@ public static double round(@SqlType(StandardTypes.DOUBLE) double num, @SqlType(S
         }
 
         double factor = Math.pow(10, decimals);
-        if (num < 0) {
-            return -(Math.round(-num * factor) / factor);
+        try {
+            if (num < 0) {
+                return -(roundToLong(-num * factor, HALF_UP) / factor);
```

Comment:
do you know what happens if num * factor or -num * factor overflows double?  if it's not easy to avoid, we should at least document any limitations here.

Reply:
It should have already been handled: if the product is +/-infinity, roundToLong will throw and we'll use BigDecimal here eventually.